File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        store = set(nums)
        for num in store:
            if num - 1 not in store:
                length = 1
                while (num + length) in store:
                    length += 1
                res = max(res, length)
        return res


        

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int res = 0;
        int l = 0;
        HashMap<Character, Integer> mp = new HashMap<>();

        for(int r = 0; r < s.length(); r++){
            if(mp.containsKey(s.charAt(r))){
                l = Math.max(mp.get(s.charAt(r)) + 1, l);
            }
            mp.put(s.charAt(r), r);
            res = Math.max(res, r - l + 1);
        }
        return res;
    }
}
